lib/request_via/client.rb

```diff
@@ -3,7 +3,12 @@
 module RequestVia
 
   class Client
-    attr_reader :address, :net_http
+    attr_accessor :address
```

Comment:
This little change affects the immutability principle because you are mutating the object state.

Look this http://dustinzeisler.com/refactoring/2017/12/11/refactor-towards-immutable-objects-ruby.html article/video to see how do you can achieve the same result keeping an immutable operation.